nucypher/cli/commands/alice.py

```diff
@@ -96,6 +96,12 @@ def __init__(
         self.discovery_port = discovery_port
         self.registry_filepath = registry_filepath
         self.middleware = middleware
+        self.poa = poa
+        self.light = light
+        self.m = m
+        self.n = n
+        self.rate = rate
```

Comment:
This implies all these values are going to be applicable for all Alice commands. I also thought about doing something like this in my last PR, but in the end I decided against it because most of commands don't need them.

Reply:
Yeah - this is indeed a bit awkward, I am having a hard time grouping the grant options more locally while still having the proper state to construct a configuration.  Perhaps @fjarri can take a look?

Reply:
At the end of the day I'm trying to follow the configuration pattern from the other modules here, since m, n, rate are configuration values.

Reply:
I don't think adding these options to the group is justified. `poa`, `light` and `rate` are only used by the `init` command, and `m` and `n` used in `grant` are technically not the same ones used to generate the config (that is, in `grant` they are not used for `generate()`, but for a different purpose).

Reply:
> This implies all these values are going to be applicable for all Alice commands.

How so?

Reply:
```
└─ $ nucypher alice run --help
Usage: nucypher alice run [OPTIONS]

  Start Alice's web controller.

Options:
[...some lines removed...]
  --duration-periods FLOAT        Policy duration in periods
  --rate INTEGER RANGE            Policy rate per period (in wei)
  --n INTEGER                     N-Total KFrags
  --m INTEGER                     M-Threshold KFrags
  --light                         Indicate that node is light
  --poa                           Inject POA middleware
[...some lines removed...]
```
It happens the same for other commands, such as `public-keys` or `destroy`, which doesn't have much sense.

Reply:
I don't get your assertion "_The only place these options are implemented is in init and grant_". These options are present in `AliceConfigOptions`, which in turn is part of `AliceCharacterOptions`, which are present in most of the commands via the `@group_character_options` decorator. 

Having said that, I was in need for having the `--light` option also present during granting, so Alice could decide to run as light, overriding her config if necessary.